addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/hardware/FritzahaWebInterface.java

```diff
@@ -300,12 +315,49 @@ public FritzahaContentExchange asyncPost(String path, String args, FritzAhaCallb
         if (!isAuthenticated()) {
             authenticate();
         }
-
         FritzahaContentExchange postExchange = new FritzahaContentExchange(callback);
-
-        this.asyncclient.newRequest(this.getURL(path)).timeout(this.config.getAsyncTimeout(), TimeUnit.SECONDS)
+        asyncclient.newRequest(getURL(path)).timeout(this.config.getAsyncTimeout(), TimeUnit.SECONDS)
                 .method(HttpMethod.POST).onResponseSuccess(postExchange).onResponseFailure(postExchange) // .onComplete(postExchange)
-                .content(new StringContentProvider(this.addSID(args), "UTF-8")).send(postExchange);
+                .content(new StringContentProvider(addSID(args), "UTF-8")).send(postExchange);
         return postExchange;
     }
+
+    public FritzahaContentExchange getTemperature(String ain) {
+        return null;
```

Comment:
So this is some work in progress code? Can you exclude it from this PR? I'm not familar with the technology behind the FRITZ! binding so please discuss things like this with @robbyb67.